rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -246,14 +248,28 @@ private void processBlockMessage(@Nonnull final MessageSender sender, @Nonnull f
         }
 
         long start = System.nanoTime();
+
         BlockProcessResult result = this.blockProcessor.processBlock(sender, block);
 
         long time = System.nanoTime() - start;
-        if(time >= 1000000000)
+
+        if (time >= 1000000000)
             result.logResult(block.getShortHash(), time);
 
         Metrics.processBlockMessage("blockProcessed", block, sender.getNodeID());
 
+        long currentTimeMillis = System.currentTimeMillis();
+
+        if (result.anyImportedBestResult())
+            lastImportedBestBlock = currentTimeMillis;
+
+        if (currentTimeMillis - lastImportedBestBlock > WAIT_TIME_ACCEPT_ADVANCED_BLOCKS)
```

Comment:
@marian151 we appreciate your report about the faucet, but you're reporting in places that don't have to do anything with this.